packages/turbo-workspaces/src/managers/pnpm.ts

```diff
@@ -0,0 +1,199 @@
+  packageJson.packageManager = `${to.name}@${to.version}`;
+  logger.rootStep(
+    `adding "packageManager" field to ${project.name} root "package.json"`
+  );
+  logger.rootStep(`adding "pnpm-workspace.yaml"`);
+
+  // write the changes
+  if (!options?.dry) {
+    fs.writeJSONSync(project.paths.packageJson, packageJson, { spaces: 2 });
+    fs.writeFileSync(
+      path.join(project.paths.root, "pnpm-workspace.yaml"),
+      `packages:\n${project.workspaceData.globs
+        .map((w) => `  - "${w}"`)
+        .join("\n")}`
+    );
+  }
+
+  // root dependencies
+  updateDependencies({
+    workspace: { name: "root", paths: project.paths },
+    project,
+    to,
+    logger,
+    options,
+  });
+
+  logger.workspaceHeader();
+  project.workspaceData.workspaces.forEach((workspace) =>
+    updateDependencies({ workspace, project, to, logger, options })
+  );
+}
+
+/*
+Remove pnpm workspace data
+
+Cleaning up from pnpm involves:
+  1. Removing the pnpm-workspace.yaml file
+  2. Removing the package-lock.json file
+*/
+async function remove(args: RemoveArgs): Promise<void> {
+  const { project, logger, options } = args;
+
+  logger.mainStep(`Removing pnpm workspaces`);
+  if (project.paths.workspaceConfig) {
+    logger.subStep(`removing "pnpm-workspace.yaml"`);
+    if (!options?.dry) {
+      fs.removeSync(project.paths.workspaceConfig);
+    }
+  }
+
+  if (!options?.dry) {
+    // collect all workspace node_modules directories
+    const allModulesDirs = [
+      project.paths.nodeModules,
+      ...project.workspaceData.workspaces.map((w) => w.paths.nodeModules),
+    ];
+
+    try {
+      logger.subStep(`removing "node_modules"`);
+      await Promise.all(
+        allModulesDirs.map((dir) => fs.rm(dir, { recursive: true }))
+      );
+    } catch (err) {
+      // only throw here if we find an error other than ENOENT (dir doesn't exist)
+      if (
+        err &&
+        typeof err === "object" &&
+        "code" in err &&
+        err.code !== "ENOENT"
+      ) {
+        throw new ConvertError("Failed to remove node_modules");
+      }
+    }
+  }
+}
+
+async function clean(args: CleanArgs): Promise<void> {
+  const { project, logger, options } = args;
+
+  logger.subStep(
+    `removing ${path.relative(project.paths.root, project.paths.lockfile)}`
+  );
+  if (!options?.dry) {
+    fs.removeSync(project.paths.lockfile);
+  }
+}
+
+// converts existing, non-pnpm lockfile to a pnpm lockfile
+async function convertLock(args: ConvertArgs): Promise<void> {
+  const { project, logger, options } = args;
+
+  if (project.packageManager !== "pnpm") {
+    logger.subStep(
+      `converting ${path.relative(
+        project.paths.root,
+        project.paths.lockfile
+      )} to pnpm-lock.yaml`
+    );
+    if (!options?.dry && fs.existsSync(project.paths.lockfile)) {
+      try {
+        await execa("pnpm", ["import"], {
+          stdio: "ignore",
+          cwd: project.paths.root,
+        });
+      } catch (err) {
```

Comment:
Nit: `err` and `e` used interchangeably. global replace pretty please?

```suggestion
      } catch (err) {
```